From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Sat, 4 Apr 2020 08:12:06 +0000
Subject: st/icon: Always show empty texture if both gicons are NULL

Commit 7ff7fb5d3b3ee655b1492226f68aeadac846e247 forgot to clear the
`priv->icon_texture` actor when returning from st_icon_update(), which
means we don't always switch to an empty icon if both gicon properties
are set to NULL.

Fix this and destroy the actor before returning early from
st_icon_update().

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1178

(cherry picked from commit 07deda593ab2e9ea30e3e1dca8c6bedd30e4ac0b)

Origin: upstream, 3.36.2, commit:51e9f19f2ff5d2863a23f762223596de419fc00c
---
 src/st/st-icon.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/st/st-icon.c b/src/st/st-icon.c
index e82adcc..fbc82bb 100644
--- a/src/st/st-icon.c
+++ b/src/st/st-icon.c
@@ -419,7 +419,10 @@ st_icon_update (StIcon *icon)
     }
 
   if (priv->gicon == NULL && priv->fallback_gicon == NULL)
-    return;
+    {
+      g_clear_pointer (&priv->icon_texture, clutter_actor_destroy);
+      return;
+    }
 
   if (!st_widget_get_resource_scale (ST_WIDGET (icon), &resource_scale))
     return;
